### PR TITLE
Add AHT21 Humidity & Temperature sensor

### PR DESCRIPTION
**Pull request by afloarea**
Add AHT21 Humidity & Temperature sensor

You may have this in the works already but I got an AHT21 sensor which I found was not already in the list of supported devices so I thought I'd give this a shot and from what I can tell, it seems to be working fine.

Some additional notes:
- In my case, the sensor always returned `ok status 0x18` and the `initRegistries` call was never needed.
-  The datasheet points to an ["official website implementation"](http://aosong.com/userfiles/files/software/AHT20-21%20DEMO%20V1_3(1).rar) for the registry initialization logic, so I just adapted what was there

Thank you for maintaining this awesome project 🙏.

### CODE REVIEW

File: diozero-core/src/main/java/com/diozero/devices/AHT21.java [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+    }
+
+    /**
+     * Read both humidity and temperature.
+     *
+     * @param verifyCrc whether to check CRC byte. If CRC fails then returned value is [-1F, -1F], but if it passes
+     *                  it increases measurement confidence
+     * @return [relative humidity, temperature (Celsius)]
+     */
+    public float[] getValues(boolean verifyCrc) {
```
Comment (EAGrahamJr) [inline diozero-core/src/main/java/com/diozero/devices/AHT21.java:186]: This appears to be a lot of nested calls that are effectively identical. For example, the `verifyCrc` parameter is _always_ `true`, so it's basically not necessary.

File: diozero-core/src/main/java/com/diozero/devices/AHT21.java [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+
+        // skip status
+        // humidity is next with 20 bits
+        final int hum = (toUnsignedInt(resp[1]) << 16 | toUnsignedInt(resp[2]) << 8 | toUnsignedInt(resp[3])) >> 4;
+        // temperature is next with 20 bits
+        final int temp = (toUnsignedInt(resp[3]) & 0x0F) << 16 | toUnsignedInt(resp[4]) << 8 | toUnsignedInt(resp[5]);
+
+        // magic formula from datasheet
+        final float relativeHum = hum * 100F / 1024 / 1024;
+        final float temperature = temp * 200F / 1024 / 1024 - 50;
```
Comment (EAGrahamJr) [inline diozero-core/src/main/java/com/diozero/devices/AHT21.java:145]: Might want to put some parenthesis around relevant parts as it's hard to parse (and relying on "default" order of operations can bite you).

**Review by EAGrahamJr — COMMENTED** [review]
Hi! Hope you don't mind me commenting :grinning:

File: diozero-core/src/main/java/com/diozero/devices/AHT21.java (hunk 1 above)
Comment (afloarea) [reply diozero-core/src/main/java/com/diozero/devices/AHT21.java:186]: Thank you for the feedback.
When I implemented I considered the existing BME280 code that you already have, which has a `float[] getValues()`, `float getTemperature()` and `float getRelativeHumidty()`. The last 2 are required because of the interfaces `ThermometerInterface` and `HygrometerInterface` but I also left the `getValues` public because it can read both at the same time. So that's why there are 3 get* methods and why the they return floats.

The `parseResponse` does indeed always wrap the `readMeasurement` which I find more readable/logically split but if you think it's better in a single method, I can modify that.

For CRC, I left a flag there because while I was going through  the datasheet and sample programs, I found that AHT21 has a lot in common with AHT20, but AHT20 does not support CRC ( but I don't have a AHT20 sensor). 
But, I agree with you, I guess you would always want to verify the CRC, if available. I am not sure, now that I think about it, what should be done if the check fails, right now it returns [-1, -1], but it might be better to throw an exception since -1 is a valid temperature and the user might use the sensor through the `ThermometerInterface`. What do you think?

File: diozero-core/src/main/java/com/diozero/devices/AHT21.java (hunk 2 above)
Comment (afloarea) [reply diozero-core/src/main/java/com/diozero/devices/AHT21.java:145]: The formulas in the datasheet are:
`RH = raw / (2^20) * 100`
`T = raw / (2^20) * 200 - 50`

For me, it looks readable as it is, but leave me a reply with how you want them / what you think is better or more readable and I will update the PR. It's not clear to me where you want the parantheses.

Comment (EAGrahamJr) [reply diozero-core/src/main/java/com/diozero/devices/AHT21.java:145]: I think the _visual_ issue is you've broken up the `Math.pow(1024,2)` into two divide operations. Yes, the formulas as presented in the datasheet are clearer, but you didn't _quite_ follow it. I would also have made the numeric parts constants, with appropriate references to the formula, just because that's generally good practice.

File: diozero-core/src/main/java/com/diozero/devices/AHT21.java (hunk 1 above)
Comment (EAGrahamJr) [reply diozero-core/src/main/java/com/diozero/devices/AHT21.java:186]: Hmmm - the issue I have with following the interfaces here is it requires **two** calls to the sensor. However, trying to cache a read value is not practical, so I withdraw my suggestion about the `get`.

The CRC probably should throw an exception, since "magic error return values" (as you note) aren't necessarily applicable.

**Comment by afloarea** [comment]
Hello, I have updated the PR according to your suggestions and I await for your feedback.
Thanks

File: diozero-core/src/main/java/com/diozero/devices/AHT21.java [hunk 3]
```diff
@@ -22,6 +22,8 @@ public class AHT21 implements ThermometerInterface, HygrometerInterface {
     public static final byte WRITE_COMMAND = 0x70;
     public static final byte[] MEASURE_COMMAND = new byte[]{(byte) 0xAC, 0x33, 0x00};
 
+    private static final float TWO_TO_THE_POWER_OF_20 = 1_048_576;
```
Comment (EAGrahamJr) [inline diozero-core/src/main/java/com/diozero/devices/AHT21.java:25]:
```suggestion
    // part of the "magic number" calculation: squared(1024)
    // 
    private static final float READING_DIVISOR = 1_048_576;
```

**Review by EAGrahamJr — COMMENTED** [review]
I like, but up to Matt to actually approve. :grinning:

File: diozero-core/src/main/java/com/diozero/devices/AHT21.java [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+     * The initial value of CRC is 0xFF.
+     * CRC check type: CRC8/MAXIM
+     * Polynomial: X8+X5+X4+1 Poly：0011 0001 0x31 When placed high, it becomes 1000 1100 0x8c
+     * Adapted from
+     * <a href="http://aosong.com/userfiles/files/software/AHT20-21%20DEMO%20V1_3(1).rar">datasheet reference</a>
+     *
+     * @param data the data
+     * @return the CRC
+     */
+    private int computeCrc(byte[] data) {
```
Comment (mattjlewis) [inline diozero-core/src/main/java/com/diozero/devices/AHT21.java:108]: Please use the com.diozero.util.Crc class.

**Review by mattjlewis — CHANGES_REQUESTED** [review]
